dcc.py

```diff
@@ -372,20 +374,86 @@ def compile_dex(apkfile, filtercfg):
                 errors.append('%s:%s' % (full_name, str(e)))
                 continue
 
-            if code:
-                compiled_method_code[method_triple] = code
+            if code[0]:
+                compiled_method_code[method_triple] = code[0]
+                native_method_prototype[jni_longname] = code[1]
 
     return compiled_method_code, errors
 
 def is_apk(name):
     return name.endswith('.apk')
 
-def dcc_main(apkfile, filtercfg, outapk, do_compile=True, project_dir=None, source_archive='project-source.zip'):
+def write_empty_dynamic_register(project_dir, compiled_methods):
+    source_dir = os.path.join(project_dir, 'jni', 'nc')
+    if not os.path.exists(source_dir):
+        os.makedirs(source_dir)
+
+    filepath = os.path.join(source_dir, 'DynamicRegister.cpp')
+    with open(filepath, 'w') as fp:
+        fp.write('#include "DynamicRegister.h"\n\nconst char *DynamicRegister(JNIEnv *env) { return nullptr; }')
+
+def write_dynamic_register(project_dir, compiled_methods):
+    source_dir = os.path.join(project_dir, 'jni', 'nc')
+    if not os.path.exists(source_dir):
+        os.makedirs(source_dir)
+
+    exportList = {}
```

Comment:
code style: use snake_case : export_list
```suggestion
    export_functions = {}
```